libs/wingc/src/type_check.rs

```diff
@@ -1269,58 +1273,74 @@ impl<'a> TypeChecker<'a> {
 					}
 				};
 
-				// Type check args against constructor
-				let constructor_type = match class_env.lookup(
-					&Symbol {
-						name: WING_CONSTRUCTOR_NAME.into(),
-						span: class_symbol.span.clone(),
-					},
-					None,
-				) {
-					Ok(v) => v.as_variable().expect("Expected constructor to be a variable").type_,
-					Err(type_error) => {
-						self.type_error(type_error);
-						return self.types.anything();
-					}
-				};
-				let constructor_sig = constructor_type
-					.as_function_sig()
-					.expect("Expected constructor to be a function signature");
-
-				// Verify return type (This should never fail since we define the constructors return type during AST building)
-				self.validate_type(constructor_sig.return_type, type_, exp);
-
-				if !arg_list.named_args.is_empty() {
-					let last_arg = constructor_sig.parameters.last().unwrap().maybe_unwrap_option();
-					self.validate_structural_type(&arg_list.named_args, &arg_list_types.named_args, &last_arg, exp);
+				if !has_fields && !arg_list.pos_args.is_empty() {
+					self.expr_error(
+						&arg_list.pos_args[0],
+						format!(
+							"{} should be called with no arguments.",
```

Comment:
The error message here should be the same as if you specified an `init()` with zero arguments - e.g. `Expected 0 arguments but got 1 when instantiating "A"`